### ISSUE TITLE
After the agent is trained, the reward of the "ep_info_buffer" in the "model" and the reward evaluated by the usage "evaluate_policy" are completely different

### ISSUE THREAD

**Issue by cypawa** [DLR-RM/stable-baselines3#2091]
### 🐛 Bug

I use PPO and my custom env. I've struggled for a long time with "VecMonitor" and "VecNormalize". At the same time, I was bothered for a long time for save and load. After the agent is trained, the reward of the "ep_info_buffer" in the "model" and the reward evaluated by the usage "evaluate_policy" are completely different. At first, it was indeed a problem with load and save, and then I found out that after using VecNormalize, I needed to save and load the environment along with the model so that the reward didn't drop when training again. I used "evaluate_policy" to check the average rewards before and after load, and found that they were similar, but when I wanted to print all the actions and rewards for a single episode, I found that the rewards that "evaluate_policy" got were completely different from the rewards for the 100 episodes saved in the ep_info_buffer after training or even after continuing training, There is a very large gap. I tried to reproduce the issue with "Pendulum-v1" and found that it didn't work. In "Pendulum-v1", the rewards in the ep_info_buffer are similar to the rewards in the "evaluate_policy". I found that I had to take 13,000,000 to 2,000,000 steps to get a significant success rate and a reward increase, and I spent an hour and a half training on my laptop to get this number of steps, if you want to repeat the problem, you can refer to it. I can't upload the pkl file of ENV, I can only upload the zip file of the model, but I mentioned before that only the zip file of the model is not a way to get a good reward, and the reward is equivalent to no training.

### Code example

```python

from stable_baselines3.common.evaluation import evaluate_policy
from stable_baselines3.common.env_checker import check_env
import gymnasium as gym
from gymnasium import spaces,register
import numpy as np
from stable_baselines3 import PPO
from stable_baselines3.common.env_util import make_vec_env
from stable_baselines3.common.vec_env import SubprocVecEnv, VecMonitor
from stable_baselines3.common.vec_env import VecNormalize


register(
    id="MaglevBraking-v1",
    entry_point="__main__:MaglevBrakingEnv",
    max_episode_steps=2000,
)
class MaglevBrakingEnv(gym.Env):

    def __init__(self):
        super().__init__()

        self.observation_space = spaces.Box(
            low=np.array([-30.0 - 1, -1200 - 1, -1200 - 1, -25.0], dtype=np.float32),
            high=np.array([30.0 + 1, 1200 + 1, 1200 + 1, 25.0], dtype=np.float32),
            dtype=np.float32
        )
        self.action_space = spaces.Box(
            low=np.float32(-1.0),
            high=np.float32(1.0),
            shape=(1,),
            dtype=np.float32
        )

    def reset(self, seed=None, options=None):
        self.raw_state = np.array([30, 0.0, 1200, 0.0])
        return self._get_obs(), {}
    def _get_obs(self):
        v, s_now, d, acceleration = self.raw_state
        return np.array([v, s_now, d, acceleration], dtype=np.float32)

    def step(self, action):
        process_reward = 0
        F_s = action[0]
        v, s_now, d, acceleration = self.raw_state
        acceleration  = (-(F_s + 1) * 200000) / 75000
        new_v = v + acceleration * 0.1
        new_s = s_now + v * 0.1 + 0.5 * acceleration * 0.1 ** 2
        new_d = 1200 - new_s
        terminated = False
        truncated = False
        if new_v > 30:
            process_reward = -200
            terminated = True
        elif new_s >= 1200:
            process_reward = -200
            terminated = True
        elif abs(acceleration) >= 3:
            process_reward = -200
            terminated = True
        if not terminated:
            process_reward = 0.5
        self.raw_state = np.array([new_v, new_s, new_d, acceleration])
        info = {}
        return self._get_obs(),process_reward , terminated, truncated, info


env = MaglevBrakingEnv()
check_env(env)
env = make_vec_env("MaglevBraking-v1",n_envs=2,vec_env_cls=SubprocVecEnv,)
env = VecMonitor(env)
env = VecNormalize(env, norm_obs=True, norm_reward=True)
model = PPO("MlpPolicy",env,device='cpu',verbose=1,learning_rate=3e-5)
model.learn(total_timesteps=100000)
mean_reward, std_reward = evaluate_policy(model, env=env, n_eval_episodes=100, deterministic=True)

```

### Relevant log output / Error message

```shell
E:\Python\Python310\lib\site-packages\stable_baselines3\common\vec_env\vec_monitor.py:44: UserWarning: The environment is already wrapped with a `Monitor` wrapperbut you are wrapping it with a `VecMonitor` wrapper, the `Monitor` statistics will beoverwritten by the `VecMonitor` ones.
------------------------------------------
| rollout/                |              |
|    ep_len_mean          | 222          |
|    ep_rew_mean          | -89.7        |
| time/                   |              |
|    fps                  | 1328         |
|    iterations           | 24           |
|    time_elapsed         | 74           |
|    total_timesteps      | 98304        |
| train/                  |              |
|    approx_kl            | 0.0049320403 |
|    clip_fraction        | 0.0264       |
|    clip_range           | 0.2          |
|    entropy_loss         | -1.17        |
|    explained_variance   | 0.408        |
|    learning_rate        | 3e-05        |
|    loss                 | 0.0885       |
|    n_updates            | 230          |
|    policy_gradient_loss | 8.33e-05     |
|    std                  | 0.779        |
|    value_loss           | 0.0843       |
------------------------------------------
-----------------------------------------
| rollout/                |             |
|    ep_len_mean          | 242         |
|    ep_rew_mean          | -79.7       |
| time/                   |             |
|    fps                  | 1327        |
|    iterations           | 25          |
|    time_elapsed         | 77          |
|    total_timesteps      | 102400      |
| train/                  |             |
|    approx_kl            | 0.001429409 |
|    clip_fraction        | 0.00339     |
|    clip_range           | 0.2         |
|    entropy_loss         | -1.17       |
|    explained_variance   | 0.305       |
|    learning_rate        | 3e-05       |
|    loss                 | 0.0628      |
|    n_updates            | 240         |
|    policy_gradient_loss | -0.000992   |
|    std                  | 0.776       |
|    value_loss           | 0.157       |
-----------------------------------------
mean_reward = -0.5
ep_info_buffer = deque([{'r': -74.5, 'l': 252, 't': 62.727539}, {'r': -155.0, 'l': 91, 't': 62.800539}, {'r': -13.0, 'l': 375, 't': 62.86554}, {'r': -44.0, 'l': 313, 't': 63.118541}, {'r': 18.0, 'l': 437, 't': 63.153541}, {'r': -195.0, 'l': 11, 't': 63.162539}, {'r': -109.5, 'l': 182, 't': 63.26254}, {'r': -122.0, 'l': 157, 't': 63.38654}, {'r': -196.0, 'l': 9, 't': 63.394541}, {'r': -43.5, 'l': 314, 't': 63.41154}, {'r': -170.5, 'l': 60, 't': 63.44354}, {'r': -145.0, 'l': 111, 't': 63.506539}, {'r': -147.0, 'l': 107, 't': 63.534539}, {'r': -163.5, 'l': 74, 't': 63.565539}, {'r': -83.0, 'l': 235, 't': 63.722539}, {'r': -88.0, 'l': 225, 't': 63.747539}, {'r': -135.0, 'l': 131, 't': 63.85054}, {'r': -111.5, 'l': 178, 't': 63.86554}, {'r': -151.0, 'l': 99, 't': 63.930541}, {'r': -109.5, 'l': 182, 't': 64.075539}, {'r': -119.0, 'l': 163, 't': 64.20854}, {'r': -178.0, 'l': 45, 't': 65.710539}, {'r': 39.5, 'l': 480, 't': 65.716539}, {'r': -128.5, 'l': 144, 't': 65.833539}, {'r': -33.0, 'l': 335, 't': 65.98554}, {'r': -156.5, 'l': 88, 't': 66.05654}, {'r': -58.5, 'l': 284, 't': 66.06754}, {'r': -112.5, 'l': 176, 't': 66.208539}, {'r': -159.0, 'l': 83, 't': 66.273539}, {'r': -29.0, 'l': 343, 't': 66.329539}, {'r': -149.0, 'l': 103, 't': 66.410539}, {'r': -112.0, 'l': 177, 't': 66.553539}, {'r': -1.5, 'l': 398, 't': 66.591539}, {'r': -135.5, 'l': 130, 't': 66.66454}, {'r': -163.5, 'l': 74, 't': 66.723539}, {'r': 22.5, 'l': 446, 't': 66.95554}, {'r': 0.5, 'l': 402, 't': 67.044539}, {'r': -141.5, 'l': 118, 't': 67.049539}, {'r': -153.5, 'l': 94, 't': 67.123539}, {'r': -91.5, 'l': 218, 't': 67.216539}, {'r': -110.0, 'l': 181, 't': 68.83654}, {'r': -30.5, 'l': 340, 't': 68.86954}, {'r': -183.0, 'l': 35, 't': 68.89754}, {'r': -136.5, 'l': 128, 't': 68.93954}, {'r': -20.5, 'l': 360, 't': 69.22654}, {'r': 12.5, 'l': 426, 't': 69.23754}, {'r': -141.5, 'l': 118, 't': 69.319541}, {'r': -173.0, 'l': 55, 't': 69.36454}, {'r': -103.5, 'l': 194, 't': 69.39254}, {'r': -35.0, 'l': 331, 't': 69.627539}, {'r': 7.0, 'l': 415, 't': 69.721539}, {'r': -80.0, 'l': 241, 't': 69.818539}, {'r': -90.5, 'l': 220, 't': 69.90154}, {'r': 17.5, 'l': 436, 't': 70.172539}, {'r': 12.5, 'l': 426, 't': 70.242539}, {'r': -140.0, 'l': 121, 't': 70.268539}, {'r': -36.0, 'l': 329, 't': 71.975539}, {'r': 16.5, 'l': 434, 't': 72.087539}, {'r': 8.5, 'l': 418, 't': 72.31054}, {'r': -20.5, 'l': 360, 't': 72.37354}, {'r': 26.0, 'l': 453, 't': 72.676539}, {'r': -5.5, 'l': 390, 't': 72.689539}, {'r': -72.5, 'l': 256, 't': 72.883541}, {'r': -49.5, 'l': 302, 't': 72.93354}, {'r': 8.0, 'l': 417, 't': 73.224541}, {'r': -195.0, 'l': 11, 't': 73.233543}, {'r': -8.0, 'l': 385, 't': 73.249542}, {'r': -111.0, 'l': 179, 't': 73.39254}, {'r': -72.0, 'l': 257, 't': 73.43854}, {'r': 5.5, 'l': 412, 't': 75.183542}, {'r': -197.5, 'l': 6, 't': 75.187543}, {'r': -193.0, 'l': 15, 't': 75.19954}, {'r': -7.0, 'l': 387, 't': 75.20954}, {'r': -109.5, 'l': 182, 't': 75.341539}, {'r': -79.0, 'l': 243, 't': 75.399539}, {'r': -46.0, 'l': 309, 't': 75.58554}, {'r': -199.0, 'l': 3, 't': 75.587539}, {'r': -44.0, 'l': 313, 't': 75.646539}, {'r': -90.0, 'l': 221, 't': 75.763539}, {'r': -89.0, 'l': 223, 't': 75.824539}, {'r': -92.0, 'l': 217, 't': 75.998539}, {'r': 6.5, 'l': 414, 't': 76.094539}, {'r': -185.0, 'l': 31, 't': 76.119539}, {'r': -24.5, 'l': 352, 't': 76.28454}, {'r': 12.0, 'l': 425, 't': 76.466539}, {'r': -19.0, 'l': 363, 't': 76.578539}, {'r': -131.0, 'l': 139, 't': 76.579539}, {'r': -132.0, 'l': 137, 't': 76.68854}, {'r': -27.0, 'l': 347, 't': 78.345539}, {'r': 9.0, 'l': 419, 't': 78.51654}, {'r': 10.5, 'l': 422, 't': 78.688539}, {'r': 8.0, 'l': 417, 't': 78.848539}, {'r': -73.5, 'l': 254, 't': 78.889539}, {'r': -130.0, 'l': 141, 't': 78.96154}, {'r': -133.5, 'l': 134, 't': 79.068539}, {'r': 17.5, 'l': 436, 't': 79.23754}, {'r': 13.5, 'l': 428, 't': 79.412539}, {'r': -30.0, 'l': 341, 't': 79.511539}, {'r': -127.0, 'l': 147, 't': 79.529539}, {'r': -89.5, 'l': 222, 't': 79.695539}], maxlen=100)
The difference in reward between mean_reward and ep_info_buffer is very large.
```

### System Info

- OS: Windows-10-10.0.26100-SP0 10.0.26100
- Python: 3.10.10
- Stable-Baselines3: 2.5.0
- PyTorch: 2.5.1+cu121
- GPU Enabled: True
- Numpy: 1.26.4
- Cloudpickle: 3.1.0
- Gymnasium: 1.0.0

### Checklist

- [x] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo
- [x] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/)
- [ ] I have provided a [minimal and working](https://github.com/DLR-RM/stable-baselines3/issues/982#issuecomment-1197044014) example to reproduce the bug
- [x] I have checked my env using the env checker
- [x] I've used the [markdown code blocks](https://help.github.com/en/articles/creating-and-highlighting-code-blocks) for both code and stack traces.

**Comment by cypawa** [comment]
@araffin Sorry to bother you, I changed the code example, I cut it overnight, and found that when my reward function is simpler, the less obvious my problem is, and this is the code that reproduces the problem the most after the cut. I tried to sample the observation space at random and set the reward to 1, but that doesn't reproduce the problem. Don't know if my current code meets the requirements? I checked the env as required on the check list, and in the meantime I've been looking through the issues and docs for the past few days, and found that most of the similar issues are problems in load and save, which don't match my problems. If that doesn't suffice, can you give me some advice on where to find a problem on my own, or where I can find some technical support? Thank you for your reply.

**Comment by araffin** [comment]
https://github.com/DLR-RM/stable-baselines3?tab=readme-ov-file#maintainers
